service/src/main/java/com/ge/predix/acs/attribute/connector/management/AttributeConnectorServiceImpl.java

```diff
@@ -20,19 +18,17 @@
     @Autowired
     private ZoneResolver zoneResolver;
 
-    private final ConnectorConverter connectorConverter = new ConnectorConverter();
-
     @Override
     public boolean upsertResourceConnector(final AttributeConnector connector) {
         ZoneEntity zoneEntity = this.zoneResolver.getZoneEntityOrFail();
         validateConnectorConfigOrFail(connector);
 
-        AttributeConnectorEntity connectorEntity = zoneEntity.getResourceAttributeConnector();
-        boolean isCreated = null == connectorEntity;
-        
-        connectorEntity = connectorConverter.toConnectorEntity(connector);
-        zoneEntity.setResourceAttributeConnector(connectorEntity);
+        boolean isCreated = false;
         try {
+            AttributeConnector existingConnector = zoneEntity.getResourceAttributeConnector();
+            isCreated = (null == existingConnector);
```

Comment:
Line 29: does not look right to me, if existingConnector is not null then isCreated will be false.